### PR TITLE
Integration-based energy likelihood

### PR DESCRIPTION
**Pull request by specktakel**
Implements a new MarginalisedEnergyLikelihood which is based on analytical integration rather than simulation.
Minor fixes to the logger.
Adds a progress bar to the simulation again.

### CODE REVIEW

File: icecube_tools/point_source_likelihood/point_source_likelihood.py [hunk 1]
```diff
@@ -376,6 +387,7 @@ def _minimize(self):
         """
 
         init_index = self._energy_likelihood._min_index + (self._max_index - self._energy_likelihood._min_index)/2
+        init_index = 2.19
```
Comment (cescalara) [inline icecube_tools/point_source_likelihood/point_source_likelihood.py:390]: Probably just want L389?

File: icecube_tools/point_source_likelihood/point_source_likelihood.py [hunk 2]
```diff
@@ -393,7 +405,7 @@ def _minimize(self):
         )
         # m.fixed["index"] = True
         m.limits["ns"] = (self._ns_min, self._ns_max)
-        m.errors["index"] = 0.1
+        m.errors["index"] = 0.1242
```
Comment (cescalara) [inline icecube_tools/point_source_likelihood/point_source_likelihood.py:408]: Where does this number come from?

**Review by cescalara — APPROVED** [review]

File: icecube_tools/point_source_likelihood/point_source_likelihood.py (hunk 2 above)
Comment (specktakel) [reply icecube_tools/point_source_likelihood/point_source_likelihood.py:408]: I think from testing/printing the input, some number I might recognise more easily

File: icecube_tools/point_source_likelihood/point_source_likelihood.py (hunk 1 above)
Comment (specktakel) [reply icecube_tools/point_source_likelihood/point_source_likelihood.py:390]: Right, I'll change it with the next PR

**Pull request merged**